detector/muonSystem/muonSystemMuRWELL_o1_v01.cpp

```diff
@@ -0,0 +1,941 @@
+
+  //  int numCopiesY = sideLength / (2 * dimensions.y() - overlapY) + 1;
+  //  int numCopiesZ = barrelLength / (2 * dimensions.z() - overlapZ) + 1;
+
+  //  double remainderY = std::fmod((sideLength - 2 * clearance), (2 * dimensions.y() - overlapY)) / (2 * dimensions.y());
+  double remainderZ = std::fmod((barrelLength - 2 * clearance), (2 * dimensions.z() - overlapZ)) / (2 * dimensions.z()) - (2 * clearance / dimensions.z());
+  double endcapRemainderZ = std::fmod((endcapDetectorYLength - 2 * clearance), (2 * dimensions.z() - overlapZ)) / (2 * dimensions.z()) - (2 * clearance / dimensions.z());
+  //  double remainderY2 = std::fmod((sideLength2 - 2 * clearance), (2 * dimensions.y() - overlapY)) / (2 * dimensions.y());
+
+  //  double remainderYLength = remainderY * 2 * dimensions.y();
+  //  double remainderZLength = remainderZ * 2 * dimensions.z();
+
+  //  double totalOverlapValueY = (numCopiesY - 1) * overlapY;
+  //  double totalOverlapValueZ = (numCopiesZ - 1) * overlapZ;
+
+  double sideEnvX = detectorVolumeThickness/2.0; // 
+  double sideEnvY = (sideLength / 2.0); // + clearance * 15;
+  double sideEnvZ = (barrelLength / 2.0); // + clearance/2.0;
+  double sideEnvY2 = (sideLength2 / 2.0);
+
+  //  double barrelRadiatorEnvX = barrelRadiatorThickness/2.0;
+  //  double barrelRadiatorEnvY = barrelRadiatorSideLength2 / 2.0;
+  //  double barrelRadiatorEnvZ = barrelLength / 2.0;
+
+  //  double endcapRadiatorEnvX = endcapRadiatorLayerOuterRadius;  // it depends on outer radius
+  //  double endcapRadiatorEnvY = endcapRadiatorLayerOuterRadius;  // outer radius too
+  //  double endcapRadiatorEnvZ = endcapRadiatorThickness / 2.0;  // layer thickness
+
+  //  double endcapDetectorEnvX = endcapDetectorLayerOuterRadius;  // it depends on outer radius
+  //  double endcapDetectorEnvY = endcapDetectorLayerOuterRadius;  // outer radius too
+  double endcapDetectorEnvZ = detectorVolumeThickness;  // layer thickness
+
+  //  double sideY = (numCopiesY -1) * 2 * dimensions.y() + remainderYLength - totalOverlapValueY;
+  //  double sideZ = (numCopiesZ -1) * 2 * dimensions.z() + remainderZLength - totalOverlapValueZ;
+
+  //  int nameCounter = 0;
+  int barrelIdCounter = 1;
+  int endcapIdCounter = 1;
+
+// ----------------------------------------------------------------------------------------------------
+//                               // B A R R E L //
+// ----------------------------------------------------------------------------------------------------
+
+  if (numBarrelDetectorLayers > 0){
+
+  int numBarrelLayer = 1;
+  double barrelLayerRMin = radius - detectorVolumeThickness/2.0;   // have to be changed depends on xml impementation
+  double barrelLayerRMax = radius + 1.5 * detectorVolumeThickness;
```

Comment:
Is this 1.5 factor always correct?

Reply:
Yes, since the detector volume was centered in the 0.25 of the thickness of that volume. But I have also changed it in the new commits, and I centered it in the middle instead.